docker/faabric-base.dockerfile

```diff
@@ -1,18 +1,48 @@
-FROM ubuntu:20.04
+FROM ubuntu:22.04
 
-RUN apt-get update
-RUN apt-get install -y software-properties-common gpg wget curl
-RUN wget -O - https://apt.llvm.org/llvm-snapshot.gpg.key|apt-key add -
-RUN wget -O - https://apt.kitware.com/keys/kitware-archive-latest.asc | apt-key add -
-RUN add-apt-repository -y -n "deb http://apt.llvm.org/focal/ llvm-toolchain-focal-10 main"
-RUN add-apt-repository -y -n "deb http://apt.llvm.org/focal/ llvm-toolchain-focal-13 main"
-RUN add-apt-repository -y -n "deb https://apt.kitware.com/ubuntu/ focal main"
-RUN add-apt-repository -y -n ppa:ubuntu-toolchain-r/test
+# Configure APT repositories
+RUN apt update \
+    && apt upgrade -y \
+    && apt install -y \
+        curl \
+        gpg \
+        software-properties-common \
+        wget \
+    # apt-key add is now deprecated for security reasons, we add individual
+    # keys manually.
+    # https://askubuntu.com/questions/1286545/what-commands-exactly-should-replace-the-deprecated-apt-key
+    && wget -O /tmp/llvm-snapshot.gpg.key \
```

Comment:
Installing llvm and cmake from 20.04 repositories in 22.04 can be prone to breakage, apt sometimes starts removing the wrong packages if you force it to resolve versions from mixed distributions like this. For LLVM, 13 and 11 are in the base 22.04 repositories, there might be no other option to install llvm 10. For cmake and conan, I switched to installing them via pip in my containers as both have pip packages, and that also made them installable on non-x86 architectures (but I don't think that's important for your usecases).
Everything else looks fine to me.

Reply:
Ok. I realised LLVM 10 was not in 22.04, so I reverted to instaling _both_ 10 and 13 from 20.04 repositories. Thanks for pointing this out.